renderer/components/ConfigBox.tsx

```diff
@@ -1,77 +1,100 @@
+  stationNumber: number;
+  ipAddress: string;
+}
+
+interface DispatchProps {
+  onIPChange: (ipAddress: string) => void;
+  onFCUpdate: (config: Config) => void;
+}
+
+interface OwnProps {
+  fcAddress: string;
+  shouldShow: boolean;
+  hide: () => void;
+}
 
-const storage = remote.require('electron-json-storage');
+type Props = StateProps & DispatchProps & OwnProps;
 
-class ConfigBoxComponent extends React.Component {
-  constructor(props) {
+interface State {
+  ipAddress: string;
+  fcAddress: string;
+  stationNumber: number;
+  originalIPAddress: string;
+  originalStationNumber: number;
+  originalFCAddress: string;
+}
+
+class ConfigBoxComponent extends React.Component<Props, State> {
+  constructor(props: Props) {
     super(props);
+
     this.state = {
       ipAddress: this.props.ipAddress,
       fcAddress: this.props.fcAddress,
       stationNumber: this.props.stationNumber,
-      originalIP: this.props.ipAddress,
-      originalSN: this.props.stationNumber,
-      originalFC: this.props.fcAddress,
+      originalIPAddress: this.props.ipAddress,
+      originalStationNumber: this.props.stationNumber,
+      originalFCAddress: this.props.fcAddress,
     };
-    this.saveChanges = this.saveChanges.bind(this);
-    this.disableUploadUpdate = this.disableUploadUpdate.bind(this);
-    this.handleIpChange = this.handleIpChange.bind(this);
-    this.handleFcChange = this.handleFcChange.bind(this);
-    this.handleStationChange = this.handleStationChange.bind(this);
-    this.handleClose = this.handleClose.bind(this);
   }
 
-
   componentDidMount() {
-    storage.get('ipAddress', (err, data) => {
+    /* eslint-disable @typescript-eslint/no-explicit-any, @typescript-eslint/ban-types */
+    storage.get('ipAddress', (err: any, data: object) => {
       if (err) {
         logging.log(err);
       } else if (!_.isEmpty(data)) {
-        this.props.onIPChange(data);
+        const ipAddress = (data as { ipAddress: string }).ipAddress;
+
+        this.props.onIPChange(ipAddress);
         this.setState({
-          ipAddress: data,
-          originalIP: data,
+          ipAddress: ipAddress,
+          originalIPAddress: ipAddress,
         });
       }
     });
 
-    storage.get('fieldControl', (err, data) => {
+    storage.get('fieldControl', (err: any, data: object) => {
       if (err) {
         logging.log(err);
       } else if (!_.isEmpty(data)) {
-        const sn = data.stationNumber;
-        const ba = data.bridgeAddress;
+        const { bridgeAddress, stationNumber } = data as Config;
+
         this.setState({
-          fcAddress: ba,
-          originalFC: ba,
-          stationNumber: sn,
-          originalSN: sn,
+          fcAddress: bridgeAddress,
+          originalFCAddress: bridgeAddress,
+          stationNumber: stationNumber,
+          originalStationNumber: stationNumber,
         });
       }
     });
   }
 
-  saveChanges(e) {
+  saveChanges = (e: React.FormEvent<HTMLInputElement>) => {
```

Comment:
How did you get the type for e? (I've been trying to look through the files to verify this type and I couldn't find any hints)

Reply:
Ah yea, so you can see that I imported `React` types that correspond to in this case a form element. So you would have to see which component calls the function and update with the type accordingly. Hopefully that clarifies a little bit lol.